src/main/java/com/employeesService/EmployeesService/service/EmployeeServiceImpl.java

```diff
@@ -46,7 +71,17 @@ public Employee update(Long empNo, Employee employee) {
     }
 
     @Override
+    @Transactional
     public void delete(Long empNo) {
         employeeRepository.deleteById(empNo);
     }
+
+    @Override
+    @Transactional
+    public void deleteEmployeeFromDepartment(String deptNo, Long empNo, LocalDate toDate) {
+        Department department = this.departmentRepository.findById(deptNo).orElseThrow(() -> new DepartmentNotFoundException(deptNo));
+        DepartmentEmployee foundDepartmentWithEmployees = this.departmentRepository.findDepartmentWithEmployees(deptNo, empNo).orElseThrow(() -> new DepartmentNotFoundException(deptNo));
```

Comment:
Here I would suggest doing something like this, as the `findDepartmentWithEmployees` is not a good approach also the name is wrong!

1. Create the `DepartmentEmployeeId` from `deptNo` and `empNo`
2. Create a method on `Department` like `findDepartmentEmployeeById()` and pass the departmentEmployeeId, the return type should be `Optional<DepartmentEmployee>`
3. If found, then set the `toDate` with the value given on the method parameter

Or it can be even better:

1. Create a method on `Department` `deleteEmployee(Long empNo, LocalDate toDate)`
2. Inside that method, implement the logic: create the DepartmentEmpoyeeId (you have all data there), find the appropriate `DepartmentEmployee` object, update it's `toDate` with the given value

Then when you call `this.departmentRepository.save(department)` will do the rest :)